### PR TITLE
Added tests for filtering in combination with paging.

### PR DESCRIPTION
**Pull request by PascalSenn**

### CODE REVIEW

File: src/HotChocolate/Data/test/Data.Filters.InMemory.Tests/FilteringAndPaging.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            res1.MatchSnapshot("true");
+
+            IExecutionResult res2 = await tester.ExecuteAsync(
+                QueryRequestBuilder.New()
+                    .SetQuery("{ root(where: { bar: { eq: false}}){ nodes { bar }}}")
+                    .Create());
+
+            res2.MatchSnapshot("false");
+        }
+
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Filters.InMemory.Tests/FilteringAndPaging.cs:40]: formatting

File: src/HotChocolate/Data/test/Data.Filters.SqlServer.Tests/FilteringAndPaging.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            res1.MatchSqlSnapshot("true");
+
+            IExecutionResult res2 = await tester.ExecuteAsync(
+                QueryRequestBuilder.New()
+                    .SetQuery("{ root(where: { bar: { eq: false}}){ nodes { bar }}}")
+                    .Create());
+
+            res2.MatchSqlSnapshot("false");
+        }
+
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Filters.SqlServer.Tests/FilteringAndPaging.cs:39]: formatting

File: src/HotChocolate/Data/test/Data.Filters.SqlServer.Tests/SchemaCache.cs [hunk 3]
```diff
@@ -5,18 +5,20 @@
 namespace HotChocolate.Data.Filters
 {
     public class SchemaCache
-        : FilterVisitorTestBase
-        , IDisposable
+        : FilterVisitorTestBase,
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Filters.SqlServer.Tests/SchemaCache.cs:8]: comma

File: src/HotChocolate/Data/test/Data.Filters.InMemory.Tests/FilteringAndPaging.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+using Snapshooter.Xunit;
+using Xunit;
+
+namespace HotChocolate.Data
+{
+    public class FilteringAndPaging
+    {
+        private static readonly Foo[] _fooEntities =
+        {
+            new Foo { Bar = true }, new Foo { Bar = false }
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Filters.InMemory.Tests/FilteringAndPaging.cs:13]: Formattting

**Review by michaelstaib — CHANGES_REQUESTED** [review]
There are some formatting issues on this one.

**Comment by PascalSenn** [comment]
@michaelstaib  fixed the issues

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**